task-log/task_log.go

```diff
@@ -0,0 +1,50 @@
+package taskLog
+
+import (
+	"bufio"
+	"fmt"
+	"net/http"
+
+	"github.com/taskcluster/taskcluster-cli/extpoints"
+)
+
+type taskLog struct{}
+
+func init() {
+	extpoints.Register("task-log", taskLog{})
+}
+
+func (taskLog) ConfigOptions() map[string]extpoints.ConfigOption {
+	return nil
+}
+
+func (taskLog) Summary() string {
+	return "Outputs the logs for <taskID> as generated, and exits when completes."
+}
+
+func (taskLog) Usage() string {
+	usage := "Usage: taskcluster task-log <taskID>\n"
+	return usage
+}
+
+func (taskLog) Execute(context extpoints.Context) bool {
+	taskID := context.Arguments["<taskID>"].(string)
+
+	path := "https://queue.taskcluster.net/v1/task/" + taskID + "/artifacts/public/logs/live.log"
+
+	resp, err := http.Get(path)
+	if err != nil {
+		panic("Error making request to " + path)
+	}
+
+	defer resp.Body.Close()
+
+	// Read line by line for live logs.
+	// This will also print the error message for failed requests.
+	scanner := bufio.NewScanner(resp.Body)
+	for scanner.Scan() {
```

Comment:
Is there any reason to read it line by line?

What if I have a task that plays a cat video as ASCII art using VLC...
Hehe :)

I don't think terminal progress bars like the ones hg or git displays when cloning are necessarily line separated.
Why not just do: `io.Copy(os.Stdout, resp.Body)` ?
If smart you would track length too, match it against content-length if given, and use the length to cut away bytes if retrying the request... So logs don't get duplicated on retries. 


If you want to be really fancy you'll copy from `resp.Body` to an intermediate buffer and flush from that buffer to `os.Stdout` at an interval... Example here:
https://github.com/taskcluster/taskcluster-worker/blob/master/runtime/ioext/copyflush.go
If the intermediate buffer implemented Flush() this CopyAndFlush method could be used.

Oh, and you might want to look into setting stdout into raw mode... Not sure all our logs are actually in raw mode, so maybe we would have to detect that... I think some of the logs uses `\r\n` as line separator which is a good indicator of raw mode.